app/models/middleware_server.rb

```diff
@@ -5,4 +5,19 @@ class MiddlewareServer < ActiveRecord::Base
   has_many :middleware_deployments, :foreign_key => "server_id", :dependent => :destroy
   serialize :properties
   acts_as_miq_taggable
+
+  # Callback from _single_quad.html.haml to determine the icon
+  # we want to display a different icon depending of the type
+  # of server we have.
+  # See also MiddlewareServerController#listicon_image for the counterpart for lists.
+  def item_image
```

Comment:
Yes. And at least in my case (only) the model has the information needed to compute the image name.